Tiloaria/assignment3/Table.cpp

```diff
@@ -0,0 +1,113 @@
+#include <cstdio>
+#include <cstdlib>
+#include <iostream>
+#include <set>
+#include <vector>
+#include <string>
+#include <assert.h>
+#include <algorithm>
+
+using namespace std;
+
+/*Dictionary - can find words or his prefics contained inside.*/
+struct Dictionary {
```

Comment:
Consider making `Dictionary` a class rather than a struct. While there's not many semantic differences between a class and a struct, the general rule is you use structs for 'passive' objects that just store data. And once you have methods, it should be a class.

Here, it's also nicer because you can then make `dict` and `pref` `private`, you can't have private members in structs :)